benches/buffer_mut.rs

```diff
@@ -0,0 +1,56 @@
+use criterion::{criterion_group, criterion_main, Criterion};
+
+fn buffer_mut(c: &mut Criterion) {
+    #[cfg(any(target_arch = "wasm32", target_arch = "wasm64"))]
+    {
+        // Do nothing.
+        let _ = c;
+    }
+
+    #[cfg(not(any(target_arch = "wasm32", target_arch = "wasm64")))]
+    {
+        use criterion::black_box;
+        use softbuffer::{Context, Surface};
+        use winit::platform::run_return::EventLoopExtRunReturn;
+
+        let mut evl = winit::event_loop::EventLoop::new();
+        let window = winit::window::WindowBuilder::new()
+            .with_visible(false)
+            .build(&evl)
+            .unwrap();
+
+        evl.run_return(move |ev, elwt, control_flow| {
+            control_flow.set_poll();
+
+            if let winit::event::Event::RedrawEventsCleared = ev {
+                control_flow.set_exit();
+
+                let mut surface = unsafe {
+                    let context = Context::new(elwt).unwrap();
+                    Surface::new(&context, &window).unwrap()
+                };
```

Comment:
This panics when I run it on Wayland, since `resize` isn't being called before `buffer_mut`:

```
Benchmarking buffer_mut(): Warming up for 3.0000 sthread 'main' panicked at 'Must set size of surface before calling `buffer_mut()`', src/wayland/mod.rs:99:14
```

I guess for consistency we need to make sure that X (and any other backends) also panics here.